src/py/flwr/server/state/state.py

```diff
@@ -134,5 +134,9 @@ def unregister_node(self, node_id: int) -> None:
         """Remove `node_id` from state."""
 
     @abc.abstractmethod
-    def get_nodes(self) -> Set[int]:
+    def get_nodes(self, workload_id: str) -> Set[int]:
         """Retrieve all currently stored node IDs as a set."""
```

Comment:
@panh99 can you similar to how it's done in `get_task_res` and some others write a spec here? For example, we could mention as a constraint:

```
If the provided `workload_id` does not exist or has no matching nodes, an empty `Set` MUST be returned.
```

Reply:
Additionally, we can define in the spec if the `workload_id` should be generated by the state or will be passed into the state.